pkg/runner/local_runner_windows.go

```diff
@@ -25,6 +26,24 @@ func NewPlainCommandCreator(sysProcAttr *syscall.SysProcAttr) CommandCreator {
 	return func(ctx context.Context, arguments []string, inputRootDirectory *path.Builder, workingDirectoryParser path.Parser, pathVariable string) (*exec.Cmd, error) {
 		// TODO: This may not work correctly if the action sets
 		// the PATH environment variable explicitly.
+		//
+		// exec.CommandContext() has some smartness to call
+		// exec.LookPath() under the hood, which we don't want.
+		// Call it with a placeholder path, followed by setting
+		// cmd.Path and cmd.Args manually. This ensures that our
+		// own values remain respected.
+		argv0 := arguments[0]
+
+		pathBuilder, scopeWalker := path.EmptyBuilder.Join(path.VoidScopeWalker)
+		if err := path.Resolve(path.NewLocalParser(argv0), scopeWalker); err != nil {
+			return nil, err
+		}
+		argv0, err := pathBuilder.GetWindowsString()
+		if err != nil {
+			return nil, err
+		}
+
+		arguments[0] = "\\nonexistent-place-back-later"
```

Comment:
This is mutating one of the input arguments of this function. Is this safe/desirable? Considering that we call `exec.CommandContext()` with `arguments[0]` and `arguments[1:]` separately, why not just pass in the string there?

Reply:
Yes. With the shared command creator that is much better. This was an artifact from also setting the argv0 on Windows to match cmd.Path. But It could have been implemented better. Now though we do not choose to do so.